docs/CaliptraHardwareSpecification.md

```diff
@@ -0,0 +1,1536 @@
+| cio_sck_en_o        | output          | SPI clock enable                                          |
+| cio_csb_o\[1:0\]    | output          | Chip select \# (one hot, active low)                      |
+| cio_csb_en_o\[1:0\] | output          | Chip select \# enable (one hot, active low)               |
+| cio_csb_sd_o\[3:0\] | output          | SPI data output                                           |
+| cio_csb_sd_en_o     | output          | SPI data output enable                                    |
+| cio_csb_sd_i\[3:0\] | input           | SPI data input                                            |
+
+#### SPI\_HOST IP programming guide
+
+The operation of the SPI\_HOST IP proceeds in seven general steps.
+
+To initialize the IP:
+
+1. Program the CONFIGOPTS multi-register with the appropriate timing and polarity settings for each csb line.
+2. Set the desired interrupt parameters.
+3. Enable the IP.
+
+Then for each command:
+
+4. Load the data to be transmitted into the FIFO using the TXDATA memory window.
+5. Specify the target device by programming the CSID.
+6. Specify the structure of the command by writing each segment into the COMMAND register.
+
+    For multi-segment transactions, assert COMMAND.CSAAT for all but the last command segment.
+
+7. For transactions that expect to receive a reply, the data can then be read back from the RXDATA window.
+
+Steps 4-7 are then repeated for each subsequent command.
+
+#### UART
+
+Caliptra implements a UART block that can communicate with a serial device that is accessible to FW over the AHB-lite Interface. This is a configuration that the SoC opts-in by defining CALIPTRA\_INTERNAL\_UART.
+
+The UART block is composed of the uart implementation. For information, see the [UART HWIP Technical Specification](https://opentitan.org/book/hw/ip/uart/). The design provides support for a programmable baud rate. The UART block is shown in the following figure.
+
+*Figure 9: UART block*
+
+#### Operation
+
+Transactions flow through the UART block starting with an AHB-lite write to WDATA, which triggers the transmit module to start a UART TX serial data transfer. The TX module dequeues the byte from the internal FIFO and shifts it out bit by bit at the baud rate. If TX is not enabled, the output is set high and WDATA in the FIFO is queued up.
+
+The following figure shows the transmit data on the serial lane, starting with the START bit, which is indicated by a high to low transition, followed by the 8 bits of data.
+
+*Figure 10: Serial transmission frame*
+
+On the receive side, after the START bit is detected, the data is sampled at the center of each data bit and stored into a FIFO. A user can monitor the FIFO status and read the data out of RDATA.
+
+#### Configuration
+
+The baud rate can be configured using the CTRL.NCO register field. This should be set using the following equation:
+
+![](./images/Caliptra_eq_NCO.png)
```

Comment:
is there math/equation support in markdown?

Might be easier to maintain if we put those equations directly here versus modifying an image

Reply:
Agree that would be ideal. We could experiment with: https://github.blog/2022-05-19-math-support-in-markdown/ but will need to verify correct display outside github (our other potential output formats are html, pdf).